packages/tasit-identity-contract/src/GnosisSafe.test.js

```diff
@@ -0,0 +1,152 @@
+    ConfigLoader.setConfig(config);
+
+    provider = ProviderFactory.getProvider();
+
+    const anaPrivateKey =
+      "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60";
+
+    anaWallet = createFromPrivateKey(anaPrivateKey);
+
+    // Contract deployment setup with ana (accounts[0]) as the only owner
+    // To change that, edit the file "tasit-contract/3rd-parties/gnosis/scripts/2_deploy_contracts.js"
+    gnosisSafeContract = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+    erc20 = new DetailedERC20(ERC20_ADDRESS);
+  });
+
+  beforeEach("", async () => {
+    snapshotId = await createSnapshot(provider);
+
+    const etherBalance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${etherBalance}`).to.equal(`${ZERO}`);
+
+    const erc20Balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${erc20Balance}`).to.equal(`${ZERO}`);
+
+    //gnosisSafeContract.removeWallet();
+    //erc20.removeWallet();
+  });
+
+  afterEach("", async () => {
+    await revertFromSnapshot(provider, snapshotId);
+  });
+
+  it("wallet owner should deposit and withdraw some ethers", async () => {
+    anaWallet = anaWallet.connect(provider);
+    const tx = await anaWallet.sendTransaction({
+      to: GNOSIS_SAFE_ADDRESS,
+      value: ONE,
+    });
+    await provider.waitForTransaction(tx.hash);
+
+    const balanceAfterDeposit = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterDeposit}`).to.equal(`${ONE}`);
+
+    const signers = [anaWallet];
+    const toAddress = anaWallet.address;
+    const value = ONE;
+
+    gnosisSafeContract.setWallet(anaWallet);
+    const execTxAction = await gnosisSafeContract.sendEtherTransaction(
+      signers,
+      toAddress,
+      value
+    );
+    await execTxAction.waitForNonceToUpdate();
+
+    const balanceAfterWithdraw = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterWithdraw}`).to.equal(`${ZERO}`);
+  });
+
+  it("wallet owner should deposit and withdraw some ERC20 tokens", async () => {
+    erc20.setWallet(anaWallet);
+    const mintAction = erc20.mint(anaWallet.address, ONE);
+    await mintAction.waitForNonceToUpdate();
+
+    const transferAction = erc20.transfer(GNOSIS_SAFE_ADDRESS, ONE);
+    await transferAction.waitForNonceToUpdate();
+
+    const balanceAfterDeposit = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterDeposit}`).to.equal(`${ONE}`);
+
+    const signers = [anaWallet];
+    const tokenAddress = erc20.getAddress();
+    const toAddress = anaWallet.address;
+    const value = ONE;
+
+    gnosisSafeContract.setWallet(anaWallet);
+    const execTxAction = await gnosisSafeContract.sendERC20Transaction(
+      signers,
+      tokenAddress,
+      toAddress,
+      value
+    );
+    await execTxAction.waitForNonceToUpdate();
+
+    const balanceAfterWithdraw = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterWithdraw}`).to.equal(`${ZERO}`);
+  });
+
```

Comment:
It could be interesting to add a new test where the signer signs an M of N transaction but it isn't sent yet - perhaps we can create an issue for this

Reply:
Hmmm
Is that related to https://github.com/tasitlabs/TasitSDK/issues/162 ?

Reply:
It's somewhat related, but what I meant here was that the transaction from the multisig wasn't sent yet because there aren't enough signatures, but one transaction to sign it was sent (or wait, is the signing done via a tx or no?)

So the M-of-N nature of the multisig makes it different than #162, I think

Reply:
I'm not a Gnosis Safe expert yet :rofl: but what I've understand until now is:
- The transaction args are sent to the contract `safeContract.getTransactionHash()` and `transactionHash` is calculated.
- The signers sign the `transactionHash` (locally)
- The transaction args + signed `transactionHash` are sent to the contract `safeContract.execTransaction()` by anybody (a relayer for example).

Maybe we can create a test case with a attempt to exec a transaction with signatures < threshold. Make sense?

Reply:
But if the signers are two different ones, what does it mean for them both to sign it locally? How are the two signatures collected into one "location"? Do they all broadcast them to the same relayer?

Reply:
I like that idea for that test case

Reply:
I've meant offline when I've written "locally". 
The `transactionHash` multi-sign is the concatenation of all signs:

(`sign(signer1, txHash) + sign(signer2, txHash) + ... + sign(signerN, txHash)`) 

So, I think that somehow the sender (could be or couldn't be a signer) have to collect (or receive) concatenated or separated signatures to be able to broadcast the tx.
I don't know if I was clear.

Reply:
I've created that issue (https://github.com/tasitlabs/TasitSDK/issues/232) since the test suite has one signer for now.

Reply:
That makes sense! Thanks. Cool. I think we can resolve this then?

Reply:
Yes. Resolving.